### DISCUSSION TITLE
Many children's problems

### DISCUSSION THREAD

**Discussion by 3157696** [strongswan/strongswan#2025] [Q&A] [answered]
Hello Teams
My current connection side is openwrt using ipsec (stroke-based)
The other side is debian using swanctl (vici-based)

Auto=route in the settings of ipsec.conf and  the setting of swanctl.conf start_action=none

Why are so many child connections generated? Which parameter affects it?
192.168.1.8/29 === 192.168.100/24
How should I modify it? Thank you for your help.

> 
> root@root:~# ipsec statusall
> no files found matching '/var/ipsec/strongswan.conf'
> Status of IKE charon daemon (strongSwan 5.9.2, Linux 5.4.238, mips):
>   uptime: 15 hours, since Dec 09 07:43:47 2023
>   worker threads: 11 of 16 idle, 5/0/0/0 working, job queue: 0/0/0/0, scheduled: 2
>   loaded plugins: charon aes des sha1 md4 random nonce x509 revocation pubkey pem openssl gmp curve25519 xcbc hmac gcm curl kernel-netlink socket-default stroke updown eap-identity eap-mschapv2
> Listening IP addresses:
>   a.b.c.d
>   192.168.1.9
> Connections:
> passthrough-1:  127.0.0.1...%any  IKEv2
> passthrough-1:   local:  [127.0.0.1] uses public key authentication
> passthrough-1:   remote: uses public key authentication
> passthrough-1:   child:  192.168.1.8/29 === 192.168.1.8/29 PASS
>         2vps:  vps.domain.com...vps.domain.com  IKEv2
>         2vps:   local:  [vps.domain.com] uses public key authentication
>         2vps:    cert:  "CN=vps.domain.com"
>         2vps:   remote: [vps.domain.com] uses public key authentication
>         2vps:   child:  192.168.1.8/29 === 192.168.100/24 TUNNEL
> Shunted Connections:
> passthrough-1:  192.168.1.8/29 === 192.168.1.8/29 PASS
> Routed Connections:
>         2vps{1}:  ROUTED, TUNNEL, reqid 10
>         2vps{1}:   192.168.1.8/29 === 192.168.100/24
> Security Associations (1 up, 0 connecting):
>         2vps[6]: ESTABLISHED 59 minutes ago, a.b.c.d[vps.domain.com]...1.2.3.4[vps.domain.com]
>         2vps[6]: IKEv2 SPIs: 1e4628a6bc6c46f8_i* 575155487afc1aa8_r, public key reauthentication in 104 minutes
>         2vps[6]: IKE proposal: AES_GCM_16_256/PRF_HMAC_SHA2_384/ECP_384
>         2vps{63}:  INSTALLED, TUNNEL, reqid 10, ESP SPIs: c7373fca_i c133d0c8_o
>         2vps{63}:  AES_GCM_16_256/ECP_384, 2400 bytes_i (25 pkts, 0s ago), 5300 bytes_o (25 pkts, 1s ago), rekeying in 26 minutes
>         2vps{63}:   192.168.1.8/29 === 192.168.100/24
>         2vps{64}:  INSTALLED, TUNNEL, reqid 10, ESP SPIs: cb4cee63_i ca6bed1f_o
>         2vps{64}:  AES_GCM_16_256/ECP_384, 956 bytes_i (10 pkts, 0s ago), 1968 bytes_o (10 pkts, 1s ago), rekeying in 26 minutes
>         2vps{64}:   192.168.1.8/29 === 192.168.100/24
>         2vps{65}:  INSTALLED, TUNNEL, reqid 10, ESP SPIs: c99b5fcc_i ca27b6e6_o
>         2vps{65}:  AES_GCM_16_256/ECP_384, 2400 bytes_i (25 pkts, 0s ago), 5300 bytes_o (25 pkts, 1s ago), rekeying in 27 minutes
>         2vps{65}:   192.168.1.8/29 === 192.168.100/24
>         2vps{66}:  INSTALLED, TUNNEL, reqid 10, ESP SPIs: ceb1b915_i c944bf26_o
>         2vps{66}:  AES_GCM_16_256/ECP_384, 864 bytes_i (9 pkts, 0s ago), 1908 bytes_o (9 pkts, 1s ago), rekeying in 30 minutes
>         2vps{66}:   192.168.1.8/29 === 192.168.100/24
>         2vps{67}:  INSTALLED, TUNNEL, reqid 10, ESP SPIs: c780728f_i cd287afd_o
>         2vps{67}:  AES_GCM_16_256/ECP_384, 44704 bytes_i (466 pkts, 0s ago), 97576 bytes_o (466 pkts, 1s ago), rekeying in 27 minutes
>         2vps{67}:   192.168.1.8/29 === 192.168.100/24

`

**Comment by tobiasbrunner** [comment, marked as answer]
With older versions it's often the combination of [break-before-make reauthentication](https://docs.strongswan.org/docs/5.9/config/rekeying.html#_reauthentication) and trap policies. After closing the old IKE_SA the trap policies can trigger a duplicate CHILD_SA, over time this can produce quite the number of SAs. You could switch to make-before-break reauthentication, or disable reauthentication completely and use rekeying instead.
